x/evm/keeper/grpc_query_test.go

```diff
@@ -899,3 +900,76 @@ func (suite *KeeperTestSuite) TestNonceInQuery() {
 	})
 	suite.Require().NoError(err)
 }
+
+func (suite *KeeperTestSuite) TestQueryBaseFee() {
+	var (
+		aux    sdk.Int
+		expRes *types.QueryBaseFeeResponse
+	)
+
+	testCases := []struct {
+		name            string
+		malleate        func()
+		expPass         bool
+		enableFeemarket bool
+		enableLondonHF  bool
+	}{
+		{
+			"pass - default Base Fee",
+			func() {
+				initialBaseFee := sdk.NewInt(ethparams.InitialBaseFee)
+				expRes = &types.QueryBaseFeeResponse{BaseFee: &initialBaseFee}
+			},
+			true, true, true,
+		},
+		{
+			"pass - non-nil Base Fee",
+			func() {
+				baseFee := sdk.OneInt().BigInt()
+				suite.app.FeeMarketKeeper.SetBaseFee(suite.ctx, baseFee)
+
+				aux = sdk.NewIntFromBigInt(baseFee)
+				expRes = &types.QueryBaseFeeResponse{BaseFee: &aux}
+			},
+			true, true, true,
+		},
+		{
+			"pass - nil Base Fee when london hardfork not activated",
+			func() {
+				baseFee := sdk.OneInt().BigInt()
+				suite.app.FeeMarketKeeper.SetBaseFee(suite.ctx, baseFee)
+
+				expRes = &types.QueryBaseFeeResponse{}
+			},
+			true, true, false,
+		},
+		{
+			"pass - nil Base Fee when feemarket not activated",
+			func() {
+				baseFee := sdk.ZeroInt()
+				expRes = &types.QueryBaseFeeResponse{BaseFee: &baseFee}
+			},
+			true, false, true,
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.enableFeemarket = tc.enableFeemarket
+			suite.enableLondonHF = tc.enableLondonHF
+			suite.SetupTest()
+
+			tc.malleate()
+
+			res, err := suite.queryClient.BaseFee(suite.ctx.Context(), &types.QueryBaseFeeRequest{})
+			if tc.expPass {
+				suite.Require().NotNil(res)
+				suite.Require().Equal(expRes, res, tc.name)
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+	suite.enableFeemarket = false
+	suite.enableLondonHF = true
```

Comment:
q: why do you need to change this after the tests?

Reply:
To change it back to the default ones, so it doesn't affect the test cases executed after this one, without these lines, some other test case will fail.